File: Visual Studio 2010/CSSL4DataGridGroupHeaderStyle/ColorConverter.cs
/********************************* 模块头 **********************************\
模块名:    ColorConverter.cs
项目:      CSSL4DataGridGroupHeaderStyle
Copyright (c) Microsoft Corporation.

根据 Group Name 改变 GroupHeader 的背景.

This source is subject to the Microsoft Public License.
See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
All other rights reserved.

THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\***************************************************************************/
using System;
using System.Windows.Media;
using System.Windows.Data;

namespace CSSL4DataGridGroupHeaderStyle
{
    public class ColorConverter : IValueConverter
    {
        public object Convert (Object value , Type targetType , Object parameter , System.Globalization.CultureInfo culture)
        {
            if (value.Equals("儿童"))
            {
                return new SolidColorBrush(Colors.Yellow);
            }
            if (value.Equals("成人"))
            {
                return new SolidColorBrush(Colors.Orange);
            }
            else
                return new SolidColorBrush(Colors.Gray);
        }
        public object ConvertBack (object value , Type targetType , object parameter , System.Globalization.CultureInfo culture)
        {
            throw new InvalidOperationException("Converter cannot convert back.");
        }
    }
}

File: Visual Studio 2010/VBSL4DataGridGroupHeaderStyle/ColorConverter.vb
'********************************* 模块头 **********************************\
' 模块名:    ColorConverter.vb
' 项目:      VBSL4DataGridGroupHeaderStyle
' Copyright (c) Microsoft Corporation.
' 
' 根据 Group Name 改变 GroupHeader 的背景.
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
' All other rights reserved.
' 
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'***************************************************************************/
Imports System.Windows.Media
Imports System.Windows.Data

Public Class ColorConverter
    Implements IValueConverter

    Public Function Convert(ByVal value As Object, ByVal targetType As Type, _
    ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object _
    Implements IValueConverter.Convert
        If value.Equals("儿童") Then
            Return New SolidColorBrush(Colors.Yellow)
        End If
        If value.Equals("成人") Then
            Return New SolidColorBrush(Colors.Orange)
        Else
            Return New SolidColorBrush(Colors.Gray)
        End If
    End Function

    Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, _
    ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object _
    Implements IValueConverter.ConvertBack
        Throw New InvalidOperationException("Converter cannot convert back.")
    End Function

End Class


